ow_lander/scripts/arm_action_servers.py

```diff
@@ -365,7 +364,7 @@ def handle_guarded_move_done(self, state, result):
         :type state: int
         :type result: FollowJointTrajectoryResult
         """
-        ground_detected = state == GoalStatus.PREEMPTED
+        ground_detected = state == GoalStatus.PREEMPTED and server_stop.stopped is False
```

Comment:
Seeing as this boolean expression is repeated in two different places, I'd like to suggest one of two possible refactors to reduce code repetition.
### Option 1 - Call expression once and store result in member variable
```suggestion
        self._ground_detected = state == GoalStatus.PREEMPTED and server_stop.stopped is False
```
then line 443 would be 
```Python
success = trajectory_async_executer.success() and trajectory_async_executer.wait() and self._ground_detected
```
**NOTE** The _ground_detected member variable would also need to be reset to false at the top of `on_guarded_move_action`.

### Option 2 - Make expression a member function and call twice
```suggestion
        ground_detected = self._was_ground_detected(state)
```
an example implementation of this function is
```Python
def _was_ground_detected(self, state):
        return state == GoalStatus.PREEMPTED and server_stop.stopped is False
```

### Conclusion
I feel **Option 1** is the best and simplest of the two possible refactors; however, since the boolean expression is only checked once it is procedurally different from what happens now.

Reply:
It would seem okay to defer this change to the refactoring Jira for this module, for the sake of merging this fix, which many people are tripping on.